### ISSUE TITLE
Implement basic view modifiers

### ISSUE THREAD

**Issue by stackotter** [stackotter/swift-cross-ui#3]
SwiftUI has a concept called [view modifiers](https://developer.apple.com/documentation/swiftui/viewmodifier). SwiftUI's `ViewModifier` protocol is for modifiers that can apply to any view, but individual views can also declare their own modifiers by creating a method of the form `func modifierName(/* parameters */) -> Self` which modifies the view accordingly and returns a copy.

The following view modifiers would be the most useful to have:

- [x] `padding`
- [x] `foreground` (colour)
- [x] `background` (colour)
- [ ] `scale`

If you think of any more, reply below.

~~The [SwiftGtk package](https://github.com/stackotter/SwiftGtk) currently doesn't support CSS styling which means that [the styling feature request for SwiftGtk](https://github.com/stackotter/SwiftGtk/issues/1) would have to be completed first.~~ SwiftGtk now supports CSS styling and has been merged into the SwiftCrossUI repo for ease of rapid development.

**Comment by skela** [comment]
Not sure if this belongs in here or its own task, but the lack of disabled for Buttons is a pretty obvious one that i encountered 10 seconds into looking into this library.

In Swift UI its:

```swift
Button("Whatever")
{
}
.disabled(true)
```

**Comment by bbrk24** [comment]
`disabled` is a little more complex than it appears at first because it disables all controls (buttons, text fields, etc) no matter how deep in the view hierarchy they may be. I agree that it is necessary, though.

**Comment by stackotter** [comment]
It has been on my todos for quite a while but I’ve mostly been focusing on other parts of SwiftCrossUI. Now that I’ve implemented an environment system, propagating disabledness down the view hierarchy should be easy enough. The backend implementations for disabling all the various controls would probably take an hour or something? With Windows being the annoying one cause SwiftCrossUI stuff takes quite a while to build in my Windows VM atm (due to the swift-winui library I’m using).

**Comment by stackotter** [comment]
I've just implemented `disabled` for all controls and all backends.

Commit: https://github.com/stackotter/swift-cross-ui/commit/b8496c0a4c960667eefb6969fe95afa280e9ce46

---

I'm gonna close this issue cause we definitely have 'basic view modifiers' now. The `scale` modifier I listed in the original issue isn't as simple as it first seems and isn't as common as some of the other modifiers we still need to add, so it would make more sense as its own issue if someone needs it urgently.
